vendor/k8s.io/kubernetes/pkg/kubelet/client/kubelet_client.go

```diff
@@ -159,10 +162,18 @@ func (k *NodeConnectionInfoGetter) GetConnectionInfo(nodeName types.NodeName) (*
 		return nil, err
 	}
 
-	// Find a kubelet-reported address, using preferred address type
-	host, err := nodeutil.GetPreferredNodeAddress(node, k.preferredAddressTypes)
-	if err != nil {
-		return nil, err
+	var host string
+
+	host, err = getMasterTrafficNodeIPAnnotation(node)
```

Comment:
-1 on this... why are we looking to introduce another method of specifying node address info?

Reply:
also, this doesn't match the PR description... this is master->node, not node->master

Reply:
We are supporting both master -> node and node -> master communication. Yes, this code is for master -> node communication (I should have said master<->node in the description).
Yes, Our preference was to add one more NodeAddressType for this but we are not sure if upstream will be interested in taking this change (https://github.com/kubernetes/kubernetes/issues/46664). So we are using annotation to avoid disturbing other code pieces that fetches address from node status.

Reply:
We should not introduce incompatible carries for master->node communication

Reply:
@liggitt *incompatible* you mean it causes problems during kube rebases?
Do you think we should carry new NodeAddressType for this use case instead of annotation?

Reply:
> incompatible you mean it causes problems during kube rebases?

incompatible because it means our kubelets and apiservers communicate with each other in non-standard ways, via non-standard configuration.

> Do you think we should carry new NodeAddressType for this use case instead of annotation?

No, if the use case is valid, make the case upstream for the API change.